packages/cli/src/commands/publish.ts

```diff
@@ -0,0 +1,50 @@
+// Copyright 2020-2021 OnFinality Limited authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import {lstatSync} from 'fs';
+import path from 'path';
+import {Command, flags} from '@oclif/command';
+import Build from './build';
+import {uploadToIpfs} from '../controller/publish-controller';
+
+export default class Publish extends Command {
+  static description = 'Upload this SubQuery project to IPFS';
+
+  static flags = {
+    location: flags.string({char: 'l', description: 'local folder'}),
+    ipfs: flags.string({description: 'IPFS gateway endpoint', default: 'http://localhost:5001/api/v0'}),
```

Comment:
This endpoint should default to a publicly accessible gateway. We probably need to host our own gateway.